tests/test_synced_collection.py

```diff
@@ -476,39 +499,42 @@ def test_reset(self, synced_list):
         with pytest.raises(ValueError):
             synced_list.reset(1)
 
+    @given(JSONDataStrategy)
     def test_insert(self, synced_list, testdata):
         synced_list.reset([1, 2])
         assert len(synced_list) == 2
         synced_list.insert(1, testdata)
         assert len(synced_list) == 3
         assert synced_list[1] == testdata
 
-    def test_reversed(self,  synced_list):
-        data = [1, 2, 3]
-        synced_list.reset([1, 2, 3])
-        assert len(synced_list) == 3
+    @given(st.lists(JSONDataStrategy, max_size=5))
+    def test_reversed(self,  synced_list, data):
+        synced_list.reset(data)
+        assert len(synced_list) == len(data)
         assert synced_list == data
         for i, j in zip(reversed(synced_list), reversed(data)):
             assert i == j
 
-    def test_remove(self, synced_list):
-        synced_list.reset([1, 2])
+    @given(d1=JSONDataStrategy, d2=JSONDataStrategy)
+    def test_remove(self, synced_list, d1, d2):
+        synced_list.reset([d1, d2])
         assert len(synced_list) == 2
-        synced_list.remove(1)
+        synced_list.remove(d1)
         assert len(synced_list) == 1
-        assert synced_list[0] == 2
-        synced_list.reset([1, 2, 1])
-        synced_list.remove(1)
+        assert synced_list[0] == d2
+        synced_list.reset([d1, d2, d1])
+        synced_list.remove(d1)
         assert len(synced_list) == 2
-        assert synced_list[0] == 2
-        assert synced_list[1] == 1
+        assert synced_list[0] == d2
+        assert synced_list[1] == d1
 
-    def test_call(self, synced_list):
-        synced_list.reset([1, 2])
-        assert len(synced_list) == 2
+    @given(st.lists(JSONDataStrategy, max_size=5))
+    def test_call(self, synced_list, testdata):
+        synced_list.reset(testdata)
+        assert len(synced_list) == len(testdata)
         assert isinstance(synced_list(), list)
         assert not isinstance(synced_list(), SyncedCollection)
```

Comment:
A small improvement for additional rigor.
```suggestion
        assert not isinstance(synced_list(), SyncedCollection)
        assert not isinstance(synced_list(), type(synced_list))
```